pywbemtools/pywbemcli/_connection_repository.py

```diff
@@ -255,76 +355,83 @@ def _load_connections_file(self):
                                     replace_underscores=True, **svr)
                             self._loaded = True
                     except KeyError as ke:
-                        raise KeyError("Items missing from record {0} in "
-                                       "connections file. Exception {1}".format
-                                       (ke, self._connections_file))
+                        raise ConnectionsFileLoadError(
+                            self._connections_file,
+                            "Element: {} missing from svr name: {}".format(
+                                ke, name))
                     except TypeError as te:
-                        raise TypeError('Invalid object type in connections '
-                                        'file: "{0}"; server name: "{1}". '
-                                        'Item: {2}'.
-                                        format(self._connections_file, name,
-                                               te))
-                except ValueError as ve:
-                    raise ValueError("Invalid YAML in connections file {0}. "
-                                     "Exception {1}".format
-                                     (self._connections_file, ve))
+                        raise ConnectionsFileLoadError(
+                            self._connections_file,
+                            'Invalid object type server name: "{0}". '
+                            'Item: "{1}"'.format(name, te))
+
+                except (TypeError, yaml.YAMLError) as exc:
+                    raise ConnectionsFileLoadError(
+                        self._connections_file,
+                        'Invalid YAML. Exception: {0}'.format(exc))
+
         else:
-            raise ConnectionsFileNotFoundError(
-                "Connections file does not exist: {0}".
-                format(self.connections_file))
+            raise ConnectionsFileNotFoundError(self.connections_file)
 
-    def add(self, name, svr_definition):
+    def add(self, svr_definition):
         """
         Add a new connection to the connections repository or replace an
         existing connection.  Users of this method should check before add if
         they do not want to replace an existing entry.
 
         Parameter:
 
-          svr_definition (:class:`ServerDefinition`):
-            An instance of ServerDefinition that contains the data that will be
-            added to the connections file.
+          svr_definition (:class:`PywbemServer`):
+            An instance of PywbemServer that contains the data that will be
+            added to the connections file. The server definition must contain
+            a name attribute.
 
         Raises:
-            click.Abort if the load of the connections file returns an
-            exception
+            ConnectionFileLoadError: if the file is invalid
+            ConnectionFileWriteError: if it cannot write the modified file.
```

Comment:
Missing `s` in exception class names: `ConnectionFile...` -> `ConnectionsFile...`